src/transform/src/dataflow.rs

```diff
@@ -23,8 +23,12 @@ use std::collections::{HashMap, HashSet};
 /// This method is currently limited in scope to propagating filtering and
 /// projection information, though it could certainly generalize beyond.
 pub fn optimize_dataflow(dataflow: &mut DataflowDesc) {
-    optimize_dataflow_demand(dataflow);
     optimize_dataflow_filters(dataflow);
+    // TODO: once the propagated filters are removed from the
```

Comment:
Can I remove propagated filters from the RelationExpr they originally came from now? Or is there some other code that we're waiting on?

Reply:
There is a WIP PR that is meant to grab project, filter, and map operators and bundle them up together. I was hoping to do that there as a clearer place to do things (for example, filtering, demand, etc may not be visible as operators sitting right around a `Get` AST node).

Reply:
But to check something else out in the meantime, there is a PR that pushes guaranteed predicate information upward, which is one way to remove predicates that are now known to be true. But, I think before stitching that in, we should coordinate on the right way to perform inter-view optimization.